bpf/process/bpf_process_event.h

```diff
@@ -453,6 +454,26 @@ get_current_subj_caps(struct msg_capabilities *msg, struct task_struct *task)
 	__get_caps(msg, cred);
 }
 
+static inline __attribute__((always_inline)) void
+get_current_subj_creds_uids(struct tg_cred_minimal *info, struct task_struct *task)
+{
+	const struct cred *cred;
+
+	/* Get the task's subjective creds */
+	probe_read(&cred, sizeof(cred), _(&task->cred));
+
+	probe_read(&info->uid, sizeof(__u32), _(&cred->uid));
+	probe_read(&info->gid, sizeof(__u32), _(&cred->gid));
+	probe_read(&info->euid, sizeof(__u32), _(&cred->euid));
+	probe_read(&info->egid, sizeof(__u32), _(&cred->egid));
+	probe_read(&info->suid, sizeof(__u32), _(&cred->suid));
+	probe_read(&info->sgid, sizeof(__u32), _(&cred->sgid));
+	probe_read(&info->fsuid, sizeof(__u32), _(&cred->fsuid));
+	probe_read(&info->fsgid, sizeof(__u32), _(&cred->fsgid));
+	probe_read(&info->securebits, sizeof(__u32), _(&cred->securebits));
```

Comment:
you could probably do this with a single probe_read using sizeof(offset_of()) but would make some assumptions about struct layout. Probably by its self might not matter but we might consider this type of things across all the progs we are starting to do lots of probe_reads.

Reply:
yes let me see, however for generalizing that I'm not sure as some structs may have ifdef CONFIG_** inside

Reply:
ah btw also forget to mention about struct layouts tagged to be randomized with gcc  ...

Reply:
sounds good it was sort of a micro-optimization anyways.